crates/bevy_input/src/keyboard.rs

```diff
@@ -650,3 +653,795 @@ pub enum KeyCode {
     /// General-purpose function key.
     F35,
 }
+
+/// Contains the platform-native logical key identifier
+///
+/// Exactly what that means differs from platform to platform, but the values are to some degree
+/// tied to the currently active keyboard layout. The same key on the same keyboard may also report
+/// different values on different platforms, which is one of the reasons this is a per-platform
+/// enum.
+///
+/// This enum is primarily used to store raw keysym when Winit doesn't map a given native logical
+/// key identifier to a meaningful [`Key`] variant. This lets you use [`Key`], and let the user
+/// define keybinds which work in the presence of identifiers we haven't mapped for you yet.
+#[derive(Debug, Clone, Ord, PartialOrd, PartialEq, Eq, Hash, Reflect)]
+#[cfg_attr(feature = "serialize", derive(Serialize, Deserialize))]
+pub enum NativeKey {
+    /// Unidentified
+    Unidentified,
+    /// An Android "keycode", which is similar to a "virtual-key code" on Windows.
+    Android(u32),
+    /// A macOS "scancode". There does not appear to be any direct analogue to either keysyms or
+    /// "virtual-key" codes in macOS, so we report the scancode instead.
+    MacOS(u16),
+    /// A Windows "virtual-key code".
+    Windows(u16),
+    /// An XKB "keysym".
+    Xkb(u32),
+    /// A "key value string".
+    Web(SmolStr),
+}
+
+/// The logical key code of a [`KeyboardInput`].
+///
+/// ## Usage
+///
+/// It is used as the generic `T` value of an [`Input`] to create a `Res<Input<Key>>`.
+/// The resource values are mapped to the current layout of the keyboard and correlate to a [`KeyCode`].
+///
+/// ## Updating
+///
+/// The resource is updated inside of the [`keyboard_input_system`].
+///
+/// ## Technical
+///
+/// Its values map 1 to 1 to winit's Key.
```

Comment:
Doc link here would be useful.

Reply:
We can't really "doc link", not to the exact type, because we could rely on another backend. This note is more for contributors to understand where this is coming from. I think a link to winit would be useful but maybe difficult to maintain ?